src/net/udp/dhcp.c

```diff
@@ -567,6 +568,27 @@ static void dhcp_request_rx ( struct dhcp_session *dhcp,
 	/* Record assigned address */
 	dhcp->local.sin_addr = ip;
 
+
+    /* Looking for DHCP option 121 */
+    int offset_opt121 = 0;
+    if ( ( offset_opt121 = find_dhcp_option_with_encap( &dhcppkt->options,
+            DHCP_CLASSLESS_STATIC_ROUTE, NULL ) ) < 0 ) {
+        /* Option 121 not found. Looking for Microsoft's option 249 */
+        if ( ( offset_opt121 =
+                find_dhcp_option_with_encap( &dhcppkt->options,
+                        DHCP_CLASSLESS_STATIC_ROUTE_MS, NULL ) ) < 0 ) {
+            /* Neither option 121 found nor 249 */
+            dhcp->netdev->static_routes_len = 0;
+        }
+    }
+    if ( offset_opt121 > 0 ) {
+        struct dhcp_option *d_option =
+                dhcp_option ( &dhcppkt->options, offset_opt121 );
+        DBGC ( dhcp, "DHCP option 121 length: %d\n", d_option->len );
+        dhcp->netdev->static_routes_len = d_option->len;
+        dhcp->netdev->static_routes = ( ( uint8_t * ) d_option->data );
```

Comment:
You can't record a long-lived pointer to this structure here, because you have nothing that guarantees that the thing pointed to will remain valid.